pinot-core/src/main/java/org/apache/pinot/core/plan/CombinePlanNode.java

```diff
@@ -92,27 +94,42 @@ public Operator run() {
       // Calculate the time out timestamp
       long endTime = System.currentTimeMillis() + TIME_OUT_IN_MILLISECONDS_FOR_PARALLEL_RUN;
 
-      int threads = Math.min(numPlanNodes / MIN_TASKS_PER_THREAD + ((numPlanNodes % MIN_TASKS_PER_THREAD == 0) ? 0 : 1),
-          // ceil without using double arithmetic
-          MAX_PLAN_THREADS);
-      int opsPerThread = Math.max(numPlanNodes / threads + ((numPlanNodes % threads == 0) ? 0 : 1),
-          // ceil without using double arithmetic
-          MIN_TASKS_PER_THREAD);
+      int numThreads = Math.min((numPlanNodes + NUM_PLANS_PER_THREAD - 1) / NUM_PLANS_PER_THREAD, MAX_PLAN_THREADS);
```

Comment:
while this math seems correct to me, let us avoid making changes unrelated to the PR.

Reply:
Sure, will make another PR for the math change